### DISCUSSION TITLE
fix afk fishing farms

### DISCUSSION THREAD

**Discussion by Tizooo** [PurpurMC/Purpur#776] [__Dismissed__] [answered]
Hey, i would like to see a option that fixes afk fising farms (especially the one shown in the screenshot)
it is also possible to press F3 + T while holding right click and while it loads releasing it to automatically holding the key for you so u can completely go afk. 

![Unbenannt](https://user-images.githubusercontent.com/34001316/145608979-d173c448-dc7b-41c5-9e04-51d52d343ad1.PNG)

**Comment by emielderckx** [comment]
These farms work no longer in the latest Minecraft versions. If you want us to add them back, I don't believe that's a good idea. I find it more an exploit than a feature.

**Comment by Tizooo** [comment]
in paper 1.18 they still work. did 1.18.1 fix it?

**Comment by emielderckx** [comment, marked as answer]
According to the Minecraft wiki in 1.16 the way fishing works changed, making these kinds of farms impossible:

20w12a:
Treasure can now be obtained only through fishing in open water. Open water is defined as a 5×4×5 area around the bobber, in which only air, water source blocks, or waterlogged blocks without collision can be present. Tripwire, bubble columns, and lily pads all invalidate open water.

20w13a:
Open water is now calculated differently. Each layer in the area around the bobber is checked, and each layer must have only air or water (or similar). Bubble columns no longer invalidate water layer checks and lily pads no longer invalidate air layer checks.

20w21a:
Lily Pads now are junk loot instead of treasure.

source: https://minecraft.fandom.com/wiki/Fishing#History

**Reply by Tizooo** [reply]
Well the one shown above still works for me
